website/blog/2017-04-13-1.0.0.md

```diff
@@ -0,0 +1,1062 @@
+export default function StatelessFunctionalComponent(
+  {
+    searchFilters,
+    title,
+    items
+  }
+) {
+  return <div />;
+}
+
+// After
+export default function StatelessFunctionalComponent({
+  searchFilters,
+  title,
+  items,
+}) {
+  return <div />
+}
+
+```
+
+### Add support for currying ([#1066](https://github.com/prettier/prettier/pull/1066))
+
+Redux is heavily promoting writing functions in a curried form where each
+argument is a nested function. Instead of indenting them, we're now putting them
+inline.
+
+<!-- prettier-ignore -->
+```js
+// Before
+const mw = store =>
+  next =>
+    action => {
+      return next(action)
+    };
+
+// After
+const mw = store => next => action => {
+  return next(action)
+};
+
+```
+
+### Respect escaping for JSX strings ([#1056](https://github.com/prettier/prettier/pull/1056))
+
+For strings we've tried a bunch of different solutions regarding to escaping:
+escape nothing, escape everything, escape a whitelisted set... But we couldn't
+find a heuristic that would make all the use cases reasonable. So we decided to
+leave the strings escaped the way they were inputted. We're now doing this for
+JSX strings.
+
+<!-- prettier-ignore -->
+```jsx
+// Before
+<a href="https://foo.bar?q1=foo&amp;q2=bar" />
+
+// After
+<a href="https://foo.bar?q1=foo&q2=bar" />
+
+```
+
+## Parenthesis
+
+A blessing and curse of printing from the AST is that we have to reprint all the
+parenthesis of the program. Our stance used to be to print only the minimum
+number of parenthesis that were needed for the program to be valid and execute
+the same way. Now, we're willing to add parenthesis that are not strictly needed
+but help people understand the code.
+
+### Add parenthesis for binary operators ([#1153](https://github.com/prettier/prettier/pull/1153))
+
+<!-- prettier-ignore -->
+```js
+// Before
+var sizeIndex = index - 1 >>> level & MASK;
+
+// After
+var sizeIndex = ((index - 1) >>> level) & MASK;
+
+```
+
+### Add parenthesis for no-confusing-arrow rule ([#1182](https://github.com/prettier/prettier/pull/1182))
+
+<!-- prettier-ignore -->
+```js
+// Before
+var x = a => 1 ? 2 : 3;
+
+// After
+var x = a => (1 ? 2 : 3);
+
+```
+
+### Remove parens from chained assignments ([#967](https://github.com/prettier/prettier/pull/967))
+
+<!-- prettier-ignore -->
+```js
+// Before
+this.size = (this._origin = (this._capacity = 0));
+
+// After
+this.size = this._origin = this._capacity = 0;
+
+```
+
+## Flow
+
+In the early days of prettier we've heavily focused on getting the core
+JavaScript language well supported. Now that it is in a good shape, we have more
+time to print Flow constructs in a polished way.
+
+### Add ability for flow generics to break ([#1041](https://github.com/prettier/prettier/pull/1041))
+
+<!-- prettier-ignore -->
+```ts
+// Before
+type _ReactElement<DefaultProps, Props, Config: $Diff<Props, DefaultProps>, C: $React.Component<DefaultProps, Props, any>> = $React.Element<Config>;
```

Comment:
This is the culprit of the sidebar looking weird @azz

Reply:
Isn't overflow-x set to `scroll` though?

Reply:
Yeah I have no idea